pkg/sql/plan/function/operator/op_bit.go

```diff
@@ -0,0 +1,133 @@
+// Copyright 2022 Matrix Origin
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package operator
+
+import (
+	"github.com/matrixorigin/matrixone/pkg/container/nulls"
+	"github.com/matrixorigin/matrixone/pkg/container/types"
+	"github.com/matrixorigin/matrixone/pkg/container/vector"
+	"github.com/matrixorigin/matrixone/pkg/encoding"
+	"github.com/matrixorigin/matrixone/pkg/vm/process"
+	"golang.org/x/exp/constraints"
+)
+
+type opBitT interface {
+	constraints.Unsigned
+}
+
+type opBitFun[T opBitT] func(v1, v2 T) T
+
+func opBitXor[T opBitT](v1, v2 T) T {
+	return v1 ^ v2
+}
+
+func opBitOr[T opBitT](v1, v2 T) T {
+	return v1 | v2
+}
+
+func opBitAnd[T opBitT](v1, v2 T) T {
+	return v1 & v2
+}
+
+func OpBitAndFun[T opBitT](args []*vector.Vector, proc *process.Process) (*vector.Vector, error) {
+	return OpBitBase(args, proc, args[0].GetType(), opBitAnd[T])
+}
+
+func OpBitOrFun[T opBitT](args []*vector.Vector, proc *process.Process) (*vector.Vector, error) {
+	return OpBitBase(args, proc, args[0].GetType(), opBitOr[T])
+}
+
+func OpBitXorFun[T opBitT](args []*vector.Vector, proc *process.Process) (*vector.Vector, error) {
+	return OpBitBase(args, proc, args[0].GetType(), opBitXor[T])
+}
+
+func OpBitBase[T opBitT](vectors []*vector.Vector, proc *process.Process, typ types.Type, afn opBitFun[T]) (*vector.Vector, error) {
+	left, right := vectors[0], vectors[1]
```

Comment:
Can this OpBitBase reused other operators?   Basically, what is the difference between bit_and/or with add?   You get two number and compute another.

Reply:
yes.   bit and/or/xor will use OpBitBase.

Reply:
only one difference with add/multiplication/div...
the result of bit operator will not check overflow